### PR TITLE
CFE-3955 & CFE-3954: cfengine template method now creates files by default

### PR DESCRIPTION
**Pull request by larsewi**
cfengine template method files promises now create files by default, unless `create => "false"` is specified. Files are not created if the rendering failes, unless `create => "true"` is specified.

### CODE REVIEW

**Comment by vpodzime** [comment]
Looks promising.

**Comment by larsewi** [comment]
@cf-bottom Jenkins please :)

File: tests/acceptance/10_files/01_create/cfengine_create_by_default.cf [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        template_method => "cfengine",
+        edit_template => "$(G.testfile).invalid_template",
+        if => fileexists("$(G.testfile).invalid_template");
+
+    # File should not be created even though template is valid
+      "$(G.testfile).test_4"
+        create => "false",
+        template_method => "cfengine",
+        edit_template => "$(G.testfile).valid_template",
+        if => fileexists("$(G.testfile).valid_template");
```
Comment (vpodzime) [inline tests/acceptance/10_files/01_create/cfengine_create_by_default.cf:73]: Shall we add a test with `create => "false"` and an existing file?

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise.

File: tests/acceptance/10_files/01_create/cfengine_create_by_default.cf (hunk 1 above)
Comment (larsewi) [reply tests/acceptance/10_files/01_create/cfengine_create_by_default.cf:76]: Yes 😉

**Comment by larsewi** [comment]
@cf-bottom Jenkins please :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7776/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7776/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7776/

File: cf-agent/verify_files.c [hunk 2]
```diff
@@ -574,8 +574,9 @@ static PromiseResult VerifyFilePromise(EvalContext *ctx, char *path, const Promi
         }
         else
         {
-            RecordFailure(ctx, pp, &a, "Promised to edit '%s', but file does not exist", path);
-            result = PromiseResultUpdate(result, PROMISE_RESULT_FAIL);
+            Log(LOG_LEVEL_VERBOSE,
```
Comment (olehermanse) [inline cf-agent/verify_files.c:578]: Typo in commit message: _priomises_

**Review by olehermanse — DISMISSED** [review]
Looks good. Just wait for CI 😸

**Comment by olehermanse** [comment]
@cf-bottom can you start a jenkins build now? :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7781/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7781/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7781/

**Review by vpodzime — APPROVED** [review]
ACK

**Pull request merged**